### ISSUE TITLE
Error comparing licenses in production test instance

### ISSUE THREAD

**Issue by goneall** [spdx/spdx-online-tools#208]
The following error occurs when running check license:

```
Traceback (most recent call last): File "/spdx/src/app/views.py", line 1001, 
in check_license matchingId,matchingType = utils.check_spdx_license(licensetext) File "/spdx/src/app/utils.py", line 489, 
in check_spdx_license build_spdx_licenses() File "/spdx/src/spdx-license-matcher/spdx_license_matcher/build_licenses.py", line 44, 
in build_spdx_licenses normalizeText = normalize(licenseText) NameError: global name 'normalize' is not defined
```

This occurs in the production test instance which can be accessed at http://52.32.53.255/app/check_license/

**Comment by rtgdk** [comment]
The issue is caused by this PR : https://github.com/spdx/spdx-license-matcher/pull/1/files
In `spdx_license_matcher/build_licenses.py` , imports have been removed which are used later in the file.
cc @Ugtan @goneall

**Comment by goneall** [comment]
@Ugtan Do you want to submit a PR adding the imports?

@rtgdk Thanks for identifying the error.

**Comment by Ugtan** [comment]
Sure @goneall.

**Comment by goneall** [comment]
I merged the PR - it solved the initial problem.

However, I ran into a different error when the license text did not exactly match:

```
Traceback (most recent call last): File "/spdx/src/app/views.py", line 1001, in check_license matchingId,matchingType = utils.check_spdx_license(licensetext) 
File "/spdx/src/app/utils.py", line 505, in check_spdx_license listedLicense = getListedLicense(licenseID) 
File "/spdx/src/spdx-license-matcher/spdx_license_matcher/utils.py", line 88, in getListedLicense listed_license = licenseinfofactoryclass.getListedLicenseById(licenseId) 
File "/usr/local/lib/python2.7/site-packages/jpype/_jclass.py", line 78, in _getClassFor def _getClassFor(javaClass): 
File "/usr/local/lib/python2.7/site-packages/gunicorn/workers/base.py", line 202, in handle_abort sys.exit(1) SystemExit: 1
```

**Comment by Ugtan** [comment]
@goneall The getListedLicense function uses spdx tools to get a spdx listed license from the license ID. 

Gunicorn has aborted the process for some reason. I found [this](https://stackoverflow.com/questions/52993280/why-is-gunicorn-calling-sys-exit1) on the internet. I guess changing the timeout value might help. Refer to [the](https://docs.gunicorn.org/en/stable/settings.html#timeout) link to add the value of timeout in `supervisor_api.conf` file. Let me know if it works.

**Comment by goneall** [comment]
@Ugtan Thanks for the research.  I think I know what's happening.  The first time you call the license match in Java, it loads all of the licenses from the internet which can take some time - likely causing a timeout.  I'll find a way to pre-load the licenses and see if that works.

**Comment by Ugtan** [comment]
@goneall If that's the case then let me know if you find a way to preload the license or else, we have have to find another way to build the licenses in the database.

**Comment by goneall** [comment]
@Ugtan I just updated the branch with a fix when there is a 100% match - it now works 👍 

Unfortunately, if there isn't a perfect match, I'm getting a timeout.  Before my fix, I didn't get a timeout if there wasn't a perfect match, so I'm not sure what has changed.  I suspect it may be a configuration problem, but I'm not sure.

I'll do a bit more research tomorrow.

**Comment by goneall** [comment]
Just an update - I tried this on my development docker instance and it takes 40 seconds (!)  Not sure what is is computing, but it does eventually return.

It looks like we have a performance issue with near matches.  If the match is perfect, the performance is acceptable.  If the match is way off, the performance is good.

@Ugtan if you could do some investigating, that would be great.

I also noted a related issue: https://github.com/spdx/spdx-online-tools/issues/210

**Comment by goneall** [comment]
The error was resolved with PR #212 which increases the timeout.

Added a new issue to track the performance issue #213

**Comment by Ugtan** [comment]
> It looks like we have a performance issue with near matches. If the match is perfect, the performance is acceptable. If the match is way off, the performance is good.

> @Ugtan if you could do some investigating, that would be great.

The performance issue could be because we are bound to use a bridge to run java code in python. But, I will try to look for a workaround that will help in the computation speed.
